task-3/rest-service/src/main/java/com/msashkin/controller/ProducerController.java

```diff
@@ -0,0 +1,42 @@
+package com.msashkin.controller;
+
+import com.msashkin.Message;
+import com.msashkin.eventbus.Event;
+import com.msashkin.eventbus.EventProducer;
+import com.msashkin.model.MessageRequest;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api")
+public class ProducerController {
+
+    private final EventProducer<Message<String>> helloEventProducer;
+
+    private final EventProducer<Message<String>> helloFwdEventProducer;
+
+    public ProducerController(EventProducer<Message<String>> helloEventProducer,
+                              EventProducer<Message<String>> helloFwdEventProducer) {
+        this.helloEventProducer = helloEventProducer;
+        this.helloFwdEventProducer = helloFwdEventProducer;
+    }
+
+    @PostMapping("/publish")
+    void publish(@RequestBody @Valid MessageRequest messageRequest) {
+        Message<String> message = new Message<>();
+        message.setMessage(messageRequest.getMessage());
+        message.setForwardToTopic(messageRequest.getForwardToTopic());
+
+        String topic = messageRequest.getTopic();
+        if (topic.equals("hello")) {
+            helloEventProducer.sendAsync(new Event<>(message));
```

Comment:
You can make application more reliable with using @Async controller and report the result of sendAsync() call (in case of failure, this controller may return 500 and client can re-try). 

In general - it is not good decision to use fire-and-forget approach with the application events as they should be durable.

Also you can improve reliability with using Transaction Outbox pattern (this is only for interest, it is too complex for this task): https://microservices.io/patterns/data/transactional-outbox.html